### PR TITLE
Canning apple cider and other juices

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Content "Updated Canning Recipes  to include the juices"

#### Purpose of change
Almost no fruit juices could be canned despite almost all fruit juices are able to be canned and saved. Here is one of several websites I checked out about canning fruit juice.  https://inhabitat.com/how-to-can-fruit-and-veggie-juices-to-preserve-summers-sweetness/ 

#### Describe the solution
I added canning recipes and fixed missing byproducts in some juice production.

#### Describe alternatives you've considered
Selling the cherry orchard to an industrialist and leaving Vanya behind forgotten.  

#### Testing
TBD

### CODE REVIEW

**Comment by Maleclypse** [comment]
The calorie generation tests are either failing due to randomness or something I have no control over.  The recipes are correct.

**Comment by Maleclypse** [comment]
@anothersimulacrum I've tried shrinking the component calories and that didn't make it work, I've doubled from the original calorie components and that at least reduced the number of failed tests.  I'm at a loss here and I'm not sure if the calories listed in the apple cider, oj and cranberry juice json entries are calorie overrides and that's causing the issue? because they certainly don't match up with when you actually make the item ingame.

**Comment by Maleclypse** [comment]
tincan apple
tincan cran
tincan fruit
tincan OJ
3L apple
3L cran
3L fruit
3L OJ
base apple
base cran
base fruit juice
Base OJ recipe
glassjar apple
glassjar cran
glassjar fruit
glassjar OJ
noncanned juice
sealed 3L pineapple

**Comment by Maleclypse** [comment]
Despite failed C/travis tests, everything creates fine.  There are inexplicable calorie differences between various recipes, however they tend to be around 10 kcal's except the water/fruit fruit juice recipe comes in at double the calories of pure fruit juice.

**Comment by I-am-Erk** [comment]
Move these from autolearn to a book on canning and I'll merge em

**Comment by anothersimulacrum** [comment]
@Maleclypse Oh shoot, I missed your earlier comment. I'll have a poke at these and see what I can do to get the tests passing later.

**Comment by Maleclypse** [comment]
> Move these from autolearn to a book on canning and I'll merge em

I'll make this a separate book from the rest of the canning recipes, that I'm going to probably include as a canning pot and can sealer instruction booklets in PR #37263.  This will both match how I'm finding canning books related to storing juice products online and also prevent any merge conflicts between the two PRs.  I'll also give anothersimulacrum some time to look through and see if there's anything I need to do on his end of things.  :) Thank you both!

File: data/json/recipes/food/canned.json [hunk 1]
```diff
@@ -895,12 +895,7 @@
     "batch_time_factors": [ 83, 5 ],
     "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 3 } ],
     "tools": [ [ [ "surface_heat", 200, "LIST" ] ], [ [ "pot_canning", -1 ] ] ],
-    "components": [
-      [ [ "water", 16 ], [ "water_clean", 16 ] ],
-      [ [ "jar_3l_glass", 1 ] ],
-      [ [ "sweet_fruit", 24, "LIST" ] ],
-      [ [ "sugar", 180 ] ]
-    ]
+    "components": [ [ [ "water", 16 ], [ "water_clean", 16 ] ], [ [ "jar_3l_glass", 1 ] ], [ [ "sweet_fruit", 12, "LIST" ] ] ]
```
Comment (anothersimulacrum) [inline data/json/recipes/food/canned.json]:
```suggestion
    "components": [ [ [ "water", 16 ], [ "water_clean", 16 ] ], [ [ "jar_3l_glass", 1 ] ], [ [ "sweet_fruit", 36, "LIST" ] ] ]
```

**Comment by Maleclypse** [comment]
@I-am-Erk I think the failed C++ test is not due to anything in this PR.  But I'm going to wait until you tell me this is good before I remove WIP.

**Comment by ZhilkinSerg** [comment]
Need conflict resolution.

**Comment by Maleclypse** [comment]
> Need conflict resolution.

Cool, matched it to the books itemgroup updates from yesterday.

**Pull request merged**